usecase/parser/v1_4_0_cronos/msg/msg.go

```diff
@@ -0,0 +1,294 @@
+			}
+		} else if events.HasEvent("channel_close_init") {
+			// parse msgChannelCloseInit
+			if events.GetEventByType("channel_close_init") != nil {
+				parserParams.IsEthereumTxInnerMsg = true
+				cmds, signers := ibc.ParseMsgChannelCloseInit(parserParams)
+				commands = append(commands, cmds...)
+				possibleSignerAddresses = append(possibleSignerAddresses, signers...)
+			}
+		} else if events.HasEvent("channel_close_confirm") {
+			// parse msgChannelCloseConfirm
+			if events.GetEventByType("channel_close_confirm") != nil {
+				parserParams.IsEthereumTxInnerMsg = true
+				cmds, signers := ibc.ParseMsgChannelCloseConfirm(parserParams)
+				commands = append(commands, cmds...)
+				possibleSignerAddresses = append(possibleSignerAddresses, signers...)
+			}
+		} else if events.HasEvent("connection_open_init") {
+			// parse msgConnectionOpenInit
+			if events.GetEventByType("connection_open_init") != nil {
+				parserParams.IsEthereumTxInnerMsg = true
+				cmds, signers := ibc.ParseMsgConnectionOpenInit(parserParams)
+				commands = append(commands, cmds...)
+				possibleSignerAddresses = append(possibleSignerAddresses, signers...)
+			}
+		} else if events.HasEvent("connection_open_try") {
+			// parse msgConnectionOpenTry
+			if events.GetEventByType("connection_open_try") != nil {
+				parserParams.IsEthereumTxInnerMsg = true
+				cmds, signers := ibc.ParseMsgConnectionOpenTry(parserParams)
+				commands = append(commands, cmds...)
+				possibleSignerAddresses = append(possibleSignerAddresses, signers...)
+			}
+		} else if events.HasEvent("connection_open_ack") {
+			// parse msgConnectionOpenAck
+			if events.GetEventByType("connection_open_ack") != nil {
+				parserParams.IsEthereumTxInnerMsg = true
+				cmds, signers := ibc.ParseMsgConnectionOpenAck(parserParams)
+				commands = append(commands, cmds...)
+				possibleSignerAddresses = append(possibleSignerAddresses, signers...)
+			}
+		} else if events.HasEvent("connection_open_confirm") {
+			// parse msgConnectionOpenConfirm
+			if events.GetEventByType("connection_open_confirm") != nil {
+				parserParams.IsEthereumTxInnerMsg = true
+				cmds, signers := ibc.ParseMsgConnectionOpenConfirm(parserParams)
+				commands = append(commands, cmds...)
+				possibleSignerAddresses = append(possibleSignerAddresses, signers...)
+			}
+		} else if events.HasEvent("send_packet") {
+			// parse msgTransfer
+			sendEvents := events.GetEventsByType("send_packet")
+			if len(sendEvents) > 0 {
+				parserParams.IsEthereumTxInnerMsg = true
+				cmds, signers := ibc.ParseMsgTransfer(parserParams)
+				commands = append(commands, cmds...)
+				possibleSignerAddresses = append(possibleSignerAddresses, signers...)
+			}
+		} else if events.HasEvent("create_client") {
+			// parse msgCreateClient
+			sendEvents := events.GetEventsByType("create_client")
+			if len(sendEvents) > 0 {
+				msg, err := parserParams.EvmInnerMsgDecoder.DecodeCosmosMsgFromTxInput(inputData, "MsgCreateClient")
```

Comment:
1. try to use constant rather than hard value
2. use complete offical message type e.g. `/ibc.core.client.v1.MsgCreateClient` instaed